WooCommerce/Classes/ViewRelated/Filters/FilterListViewController.swift

```diff
@@ -146,14 +155,17 @@ final class FilterListViewController<ViewModel: FilterListViewModel>: UIViewCont
     }
 
     @objc func dismissButtonTapped() {
-        dismiss(animated: true) {}
+        dismiss(animated: true) { [weak self] in
+            self?.onDismissAction()
```

Comment:
nit: it's unlikely `self` is deallocated in the dismiss completion block, but maybe we can call `onDismissAction()` before `dismiss`?

Reply:
I'd leave it in here because if I put it outside, it can generate an `unbalanced calls to begin/end appearance transitions` error if we want to present something else when the view will be dismissed. We currently do not present anything, but if this changes in the future it could generate unexpected errors.